### ISSUE TITLE
Doesn't build with newer typescript versions.

### ISSUE THREAD

**Issue by thwischm** [timbertson/shellshape#192]
Error message:

```
$ tools/gup compile
gup  compile
gup    schemas/all
gup      schemas/inputs
gup    shellshape/all
gup      build/gjs/extension.js
gup        build/gjs/.sources
../src/gjs/tiling.ts(315,6): error TS2345: Argument of type '{ item: TiledWindow; index: number; toString: () => string; }' is not assignable to parameter of type 'IndexedTiledWindow'.
  Object literal may only specify known properties, and 'toString' does not exist in type 'IndexedTiledWindow'.
gup      Target `build/gjs/extension.js` failed with exit status 2
gup    Target `shellshape/all` failed with exit status 2
gup  Target `compile` failed with exit status 2
```

Tested on Arch.

**Comment by thwischm** [comment]
It can be fixed it by specifying typescript version 2.1.5 (without the ^) in package.json, but that doesn't seem like the best idea.

**Comment by timbertson** [comment]
Thanks for the heads up, fixed 👍
